regex-assembly/exclude/unix-shell-fps-pl1.ra

```diff
@@ -0,0 +1,668 @@
+##! Please refer to the documentation at
+##! https://coreruleset.org/docs/development/regex_assembly/.
+
+##! This list excludes command words that are prone to cause false
+##! positives from the following include files:
+##! - unix-shell-upto3.ra and
+##! - unix-shell-upto3-for-cmdline.ra
+##! - unix-shell-4andup-with-params.ra
+
+##! To automatically add exclusions for English words from the sources, run the following:
+##! original="$(grep -vE '^(?:#|$)' regex-assembly/exclude/unix-shell-fps-pl1.ra)"
+##! upto3="$(grep -vE '^(?:#|$)' regex-assembly/include/unix-shell-upto3.ra)"
+##! rest="$(grep -vE '^(?:#|$)' regex-assembly/include/unix-shell-4andup-with-params.ra)"
+##! english_upto3="$(util/fp-finder/spell.sh -m -e regex-assembly/include/unix-shell-upto3.ra)"
+##! # strip suffixes from words so that spell.sh works
+##! english_rest="$(util/fp-finder/spell.sh -m -e -s '{{[^}]+}}' regex-assembly/include/unix-shell-4andup-with-params.ra)"
+##! result=""
+##! function update_existing {
+##!   if [ -z "${1}" ]; then
+##!     return
+##!   fi
+##!   while read -r oword; do
+##!     found=0
+##!     while read -r eword; do
+##!       if grep -qE "^${eword}(?:[@~]|\{\{.*|$)" <<<"${oword}"; then
+##!         result="${result}${eword}\n"
+##!         result="${result}${eword}@\n"
+##!         result="${result}${eword}~\n"
+##!         result="${result}${eword}{{space-or-redirect}}\n"
+##!         result="${result}${eword}{{no-space}}\n"
+##!         found=1
+##!         break
+##!       fi
+##!     done <<<"${1}"
+##!     if [ ${found} -eq 0 ]; then
+##!       result="${result}${oword}\n"
+##!     fi
+##!   done <<<"${original}"
+##! }
+##! function add_new {
+##!   if [ -z "${1}" ]; then
+##!     return
+##!   fi
+##!   while read -r eword; do
+##!     if ! grep -qE "^${eword}(?:[@~]|\{\{.*|$)" <<<"${original}"; then
+##!       result="${result}${eword}\n"
+##!       result="${result}${eword}@\n"
+##!       result="${result}${eword}~\n"
+##!       result="${result}${eword}{{space-or-redirect}}\n"
+##!       result="${result}${eword}{{no-space}}\n"
+##!     fi
+##!   done <<<"${1}"
+##! }
+##! update_existing "${english_upto3}"
+##! update_existing "${english_rest}"
+##! add_new "${english_upto3}"
+##! add_new "${english_rest}"
+
+##! sed -i '' -e '61,$d' regex-assembly/exclude/unix-shell-fps-pl1.ra
```

Comment:
Non GNU sed (e.g., on macOS) expects a suffix string as the argument to `-i` that will be used to write to a temporary file. If the suffix is empty, no temporary file will be created. With GNU sed the argument is optional.

Reply:
Yes, it's optional, but it can't be empty.

```
$ sed -i '' -e '61,$d' regex-assembly/exclude/unix-shell-fps-pl1.ra
sed: can't read : No such file or directory
$ sed -i -e '61,$d' regex-assembly/exclude/unix-shell-fps-pl1.ra
```

You may need to construct this differently.

Reply:
That's so annoying... I've replaced those commands with `ed`, which should be portable enough.